### PR TITLE
ifdefs on elpi version in source code

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

**Comment by gares** [comment]
@vbgl I need some nix expertise here.
It seems that `%{version:elpi}` does not work on nix, it seems to expand to the empty string.
Is it expected? It should be the version written in the META file, in turn generated by dune...

**Comment by vbgl** [comment]
I think the problem comes from the fact that `elpi` in nixpkgs is built from the git repository rather than from the released tarball (which contains correct version information, I guess when produced by dune-release).

**Comment by gares** [comment]
Hem, ok, thanks for explaining.

Next question is: how can I fix that?

Is using the tarball generated by dune-release a problem for nix?

**Comment by vbgl** [comment]
I’ll propose a patch for nixpkgs asap. What is suppose to happen to your dune scripts (in coq-elpi) when using a development version (i.e., not a release with a proper version string) of elpi?

File: etc/version_parser.ml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+let main () =
+  let v = Sys.argv.(1) in
+  let l = String.split_on_char '.' Str.(replace_first (regexp "^v") "" v) in
+  (* sanitization *)
+  let l =
+    match l with
+    | l when List.for_all is_number l -> l
+    | ["%%VERSION_NUM%%"] -> ["99";"99";"99"]
+    | _ -> Printf.eprintf "version_parser: cannot parse: %s\n" v; exit 1 in
```
Comment (gares) [inline etc/version_parser.ml:12]: @vbgl this was my "attempt" to support development versions. I'm happy to improve on that

File: etc/version_parser.ml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+let is_number x = try let _ = int_of_string x in true with _ -> false ;;
+
+let main () =
+  let v = Sys.argv.(1) in
+  let l = String.split_on_char '.' Str.(replace_first (regexp "^v") "" v) in
+  (* sanitization *)
+  let l =
+    match l with
+    | l when List.for_all is_number l -> l
+    | ["%%VERSION_NUM%%"] -> ["99";"99";"99"]
```
Comment (gares) [inline etc/version_parser.ml:11]:
```suggestion
    | ( [""] | ["%%VERSION_NUM%%"] ) -> ["99";"99";"99"]
```

**Comment by vbgl** [comment]
The fix has been merged into nixpkgs master branch. I hope this addresses you problems. Please ping again if it’s not the case. Kudos to Cyril for the patch.

**Comment by gares** [comment]
@CohenCyril opam works on master, but not nix. Since I can't run nix on my machine, is there any chance to get verbose logs in CI?

**Comment by gares** [comment]
I mean, the CI file is generated, and it contains `2>&1 > /dev/null`. Can I disable that "feature"?

**Pull request merged**